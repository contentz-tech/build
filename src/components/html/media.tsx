import { jsx } from "@emotion/core";
import Card from "../card";

function Image(props: any) {
  const img = (
    <img
      {...props}
      loading="lazy"
      css={{
        maxWidth: "100%",
        verticalAlign: "top"
      }}
    />
  );

  if (!props.title) return img;

  return (
    <Card>
      <figure
        css={{
          textAlign: "center",
          margin: 0,
          width: "100%"
        }}
      >
        {img}
        <figcaption
          css={{
            fontSize: "0.8em",
            textAlign: "center",
            marginTop: "1em",
            color: "rgba(0, 0, 0, 0.7)"
          }}
        >
          {props.title}
        </figcaption>
      </figure>
    </Card>
  );
}

export { Image };
