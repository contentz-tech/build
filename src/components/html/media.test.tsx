/** * @jest-environment jsdom */
import '@testing-library/jest-dom/extend-expect'
import { jsx } from "@emotion/core";
import { render } from "@testing-library/react";
import { Image } from "./media";

describe("Media", () => {
  test("Image without Figure", () => {
    const { asFragment, getByAltText } = render(
      <Image src="/static/avatar.png" alt="User avatar" />
    );

    expect(asFragment()).toMatchSnapshot();
    expect(getByAltText(/User avatar/)).toBeInTheDocument();
    expect(getByAltText(/User avatar/)).toHaveAttribute("loading", "lazy");
    expect(getByAltText(/User avatar/)).toHaveAttribute("loading", "lazy");
    expect(getByAltText(/User avatar/))
      .toHaveAttribute("src", "/static/avatar.png");
  });

  test("Image with Figure", () => {
    const { asFragment, getByAltText, getByText } = render(
      <Image src="/static/avatar.png" title="My Avatar" alt="User avatar" />
    );

    expect(asFragment()).toMatchSnapshot();
    expect(getByAltText(/User avatar/)).toBeInTheDocument();
    expect(getByAltText(/User avatar/)).toHaveAttribute("loading", "lazy");
    expect(getByAltText(/User avatar/))
      .toHaveAttribute("src", "/static/avatar.png");
    expect(getByText(/My Avatar/)).toBeInTheDocument();
    expect(getByText(/My Avatar/)).toHaveTextContent("My Avatar");
  });
});
