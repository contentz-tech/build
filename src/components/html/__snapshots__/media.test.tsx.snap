// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Media Image with Figure 1`] = `
<DocumentFragment>
  .emotion-0 {
  max-width: 100%;
  vertical-align: top;
}

.emotion-3 {
  box-shadow: rgba(0,0,0,0.12) 0px 2px 5px 0px;
  background-color: white;
  border-radius: 5px;
  -webkit-transition: all 0.2s ease 0s;
  transition: all 0.2s ease 0s;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  margin: 1.5em -2em 3em;
  padding: 1em 2em;
}

@media (max-width:720px) {
  .emotion-3 {
    box-shadow: none;
    border-radius: 0;
    margin: 0;
    padding: 1em 0.5em;
  }
}

.emotion-2 {
  text-align: center;
  margin: 0;
  width: 100%;
}

.emotion-1 {
  font-size: 0.8em;
  text-align: center;
  margin-top: 1em;
  color: rgba(0,0,0,0.7);
}

<div
    class="emotion-3"
  >
    <figure
      class="emotion-2"
    >
      <img
        alt="User avatar"
        class="emotion-0"
        loading="lazy"
        src="/static/avatar.png"
        title="My Avatar"
      />
      <figcaption
        class="emotion-1"
      >
        My Avatar
      </figcaption>
    </figure>
  </div>
</DocumentFragment>
`;

exports[`Media Image without Figure 1`] = `
<DocumentFragment>
  .emotion-0 {
  max-width: 100%;
  vertical-align: top;
}

<img
    alt="User avatar"
    class="emotion-0"
    loading="lazy"
    src="/static/avatar.png"
  />
</DocumentFragment>
`;
